consensus/types/src/data_column_sidecar.rs

```diff
@@ -95,23 +94,32 @@ impl<T: EthSpec> DataColumnSidecar<T> {
         for blob in blobs {
             let blob = KzgBlob::from_bytes(&blob.blob).map_err(KzgError::from)?;
             let (blob_cells, blob_cell_proofs) = kzg.compute_cells_and_proofs(&blob)?;
-            for ((column, column_proofs), (cell, proof)) in
-                iter::zip(columns.iter_mut(), column_kzg_proofs.iter_mut()).zip(iter::zip(
-                    blob_cells.into_iter(),
-                    blob_cell_proofs.into_iter(),
-                ))
-            {
+
+            // we iterate over each column, and we construct the column from "top to bottom",
+            // pushing on the cell and the corresponding proof at each column index. we do this for
+            // each blob (i.e. the outer loop).
+            for col in 0..T::number_of_columns() {
+                let cell = blob_cells
+                    .get(col)
+                    .expect("blob cell exists at index {col}");
```

Comment:
My understanding is that this is an unexpected case and there's no recovery from this, however we typically avoid `panic`s in these functions and return an `Error` and leave it for callers to handle.
In this case, if there's a bug here, proposer would miss the block rather than having the node crash.

Reply:
Updated in https://github.com/sigp/lighthouse/pull/5210/commits/75cf05f3c746fccf7fbf9c7583a760ff60e72adf